projects/js-packages/charts/src/components/line-chart/line-chart.tsx

```diff
@@ -265,7 +266,17 @@ const LineChartInternal = forwardRef< SingleChartRef, LineChartProps >(
 		);
 
 		const dataSorted = useChartDataTransform( data );
-		const { getElementStyles } = useGlobalChartsContext();
+		const { getElementStyles, isSeriesVisible } = useGlobalChartsContext();
+
+		// Filter out hidden series when using interactive legends, preserving original index
+		const visibleData = useMemo( () => {
+			if ( ! chartId || ! interactive ) {
+				return dataSorted.map( ( series, index ) => ( { series, originalIndex: index } ) );
```

Comment:
Instead of using `originalIndex`, I wonder whether it might be easier if we add a isVisbile property to series and in the loop starting L452. If a series is not visible, then we return null to skip it. With that, we wouldn't have to mess with originalIndex in there.